### PR TITLE
[SM-863] Add endpoint for fetching multiple secrets by IDs

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Code changes

* **bitwarden_license/src/Commercial.Core/SecretsManager/AuthorizationHandlers/Secrets/SecretAuthorizationHandler.cs:**
Adding CanReadSecret authorization. 

* **src/Core/SecretsManager/AuthorizationRequirements/SecretOperationRequirement.cs:**
Adding read option.

* **bitwarden_license/test/Commercial.Core.Test/SecretsManager/AuthorizationHandlers/Secrets/SecretAuthorizationHandlerTests.cs:**
Adding unit tests for the CanReadSecret option.

* **bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs:**
Bug fix since service accounts can have write access now.

* **src/Api/SecretsManager/Controllers/SecretsController.cs:**
Adding a new endpoint to fetch secrets given a list of secret IDs.

* **src/Api/SecretsManager/Models/Request/GetSecretsRequestModel.cs:**
Request model for the new endpoint.

* **src/Api/SecretsManager/Models/Response/BaseSecretResponseModel.cs:**
**src/Api/SecretsManager/Models/Response/SecretResponseModel.cs:**
Since we won't need to return access information (read, write) for the endpoint, separating out response models. 

`BaseSecretResponseModel` will just have secret information and `SecretResponseModel` will have the extra access information.
 

* **src/Core/Services/IEventService.cs:**
**src/Core/Services/Implementations/EventService.cs:**
**src/Core/Services/NoopImplementations/NoopEventService.cs:**

Adding a new method to the event service that will bulk insert many service account fetching secret events at once.

* **test/Api.IntegrationTest/SecretsManager/Controllers/SecretsControllerTests.cs:**
Integration tests for the new endpoint.

* **test/Api.Test/SecretsManager/Controllers/SecretsControllerTests.cs:**
Unit tests for the new endpoint.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: bitwarden_license/test/Commercial.Core.Test/SecretsManager/AuthorizationHandlers/Secrets/SecretAuthorizationHandlerTests.cs [hunk 1]
```diff
@@ -232,6 +232,65 @@ public void SecretOperations_OnlyPublicStatic()
...
+
+        Assert.False(authzContext.HasSucceeded);
+    }
+
+    [Theory]
+    [BitAutoData(PermissionType.RunAsAdmin, true, true, true)]
+    [BitAutoData(PermissionType.RunAsUserWithPermission, false, false, false)]
+    [BitAutoData(PermissionType.RunAsUserWithPermission, false, true, false)]
+    [BitAutoData(PermissionType.RunAsUserWithPermission, true, false, true)]
+    [BitAutoData(PermissionType.RunAsUserWithPermission, true, true, true)]
```
Comment (cd-bitwarden) [inline bitwarden_license/test/Commercial.Core.Test/SecretsManager/AuthorizationHandlers/Secrets/SecretAuthorizationHandlerTests.cs:274]: Hey Thomas, do we need to also use the PermissionType.RunAsServiceAccountWIthPermission here as well? 😄

File: test/Api.IntegrationTest/SecretsManager/Controllers/SecretsControllerTests.cs [hunk 2]
```diff
@@ -709,6 +709,69 @@ public async Task Delete_Success(PermissionType permissionType)
...
+
+        var request = new GetSecretsRequestModel { Ids = new[] { secret.Id } };
+
+        var response = await _client.PostAsJsonAsync("/secrets/get-by-ids", request);
+        Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+    }
+
+    [Theory]
+    [InlineData(PermissionType.RunAsAdmin)]
+    [InlineData(PermissionType.RunAsUserWithPermission)]
```
Comment (cd-bitwarden) [inline test/Api.IntegrationTest/SecretsManager/Controllers/SecretsControllerTests.cs:737]: Do we also need to check ServiceAccounts?

**Review by cd-bitwarden — CHANGES_REQUESTED** [review]
Great job @Thomas-Avery I just had 2 easy questions about the tests, other than that it looks great.

File: bitwarden_license/test/Commercial.Core.Test/SecretsManager/AuthorizationHandlers/Secrets/SecretAuthorizationHandlerTests.cs (hunk 1 above)
Comment (Thomas-Avery) [reply bitwarden_license/test/Commercial.Core.Test/SecretsManager/AuthorizationHandlers/Secrets/SecretAuthorizationHandlerTests.cs:274]: Good catch resolved in https://github.com/bitwarden/server/pull/3134/commits/902f25b2ceaa179883ae6a8139a5db7fad1aaf39

File: test/Api.IntegrationTest/SecretsManager/Controllers/SecretsControllerTests.cs (hunk 2 above)
Comment (Thomas-Avery) [reply test/Api.IntegrationTest/SecretsManager/Controllers/SecretsControllerTests.cs:737]: For integration tests, we don't currently have a way to login as a service account to run the tests. 

We need to build out that ability, but it is out of scope for this specific ticket.

**Review by cd-bitwarden — APPROVED** [review]
LGTM Thomas :)

**Pull request merged**